Add support for Midea Follow Me commands

carbon198 opened the pull request:
Hi there! I've added support for the follow me commands so you can send an arbitrary temperature sensor to the air handler that replaces it's inlet temperature reading. This is useful for my case because I am not using a thermostat at all, I just control the air handler with my ESP32 device. I have a temperature sensor connected to Home Assistant that I am sending as a follow me temperature reading. 
~~One odd thing about this is it forces the whole system into Celcius. Personally I have found that it operates better in Celcius anyway (I assume it was designed by people using C), and since I don't have a wall controller it doesn't affect me. However, it is something to keep in mind.~~

I also updated a few labels and byte flags based on additional research. 

Feel free to edit it or discard it, I just figured other people might benefit.

carbon198 committed:
Update debugger_method.yaml

carbon198 committed:
Update debugger_method_vars.h

carbon198 committed:
Allow Follow Me with Fahrenheit and add toggle

carbon198 commented:
I've updated the branch to fix the celcius issue--the set temp will now remain in F. If you already switched to C you will need to coerce it back with the wall controller, or you can message me and I can tell you how to do with with your esp32 device. 
There's also now a toggle for Follow Me so you can turn it off if your temperature sensor gets stale or something like that. 

Edit:
Also here are my YAMLs for anyone who wants them. I added a bunch of other stuff.
[debugger_method.yaml](https://gist.github.com/carbon198/7e6b3854d7dbc426f9b0a7c851bee871)
[debugger_method_vars.h](https://gist.github.com/carbon198/4cb35b1d102653c1f52bcfcc5cc3872c)

wtahler commented:
Thanks! Sorry it took me so long to accept, busy couple months.

carbon198 commented:
No rush my friend, it wasn't really ready until today anyway. I've been poring over the Home Assistant forums looking for all the research into this. I ended up connecting my wired controller to my esp device and just watching it's commands to figure out the Fahrenheit thing. 
I appreciate all your work on this :beers:

MrAnonymous722 commented:
> I've updated the branch to fix the celcius issue--the set temp will now remain in F. If you already switched to C you will need to coerce it back with the wall controller, or you can message me and I can tell you how to do with with your esp32 device. There's also now a toggle for Follow Me so you can turn it off if your temperature sensor gets stale or something like that.
> 
> Edit: Also here are my YAMLs for anyone who wants them. I added a bunch of other stuff. [debugger_method.yaml](https://gist.github.com/carbon198/7e6b3854d7dbc426f9b0a7c851bee871) [debugger_method_vars.h](https://gist.github.com/carbon198/4cb35b1d102653c1f52bcfcc5cc3872c)

I'm interested in kicking the unit back into F* with an ESP32.

carbon198 commented:
> > I've updated the branch to fix the celcius issue--the set temp will now remain in F. If you already switched to C you will need to coerce it back with the wall controller, or you can message me and I can tell you how to do with with your esp32 device. There's also now a toggle for Follow Me so you can turn it off if your temperature sensor gets stale or something like that.
> > Edit: Also here are my YAMLs for anyone who wants them. I added a bunch of other stuff. [debugger_method.yaml](https://gist.github.com/carbon198/7e6b3854d7dbc426f9b0a7c851bee871) [debugger_method_vars.h](https://gist.github.com/carbon198/4cb35b1d102653c1f52bcfcc5cc3872c)
> 
> I'm interested in kicking the unit back into F* with an ESP32.

Have you tried using this follow me feature? I think it should kick it back to F. If not I can test something else out.

MrAnonymous722 commented:
> > > I've updated the branch to fix the celcius issue--the set temp will now remain in F. If you already switched to C you will need to coerce it back with the wall controller, or you can message me and I can tell you how to do with with your esp32 device. There's also now a toggle for Follow Me so you can turn it off if your temperature sensor gets stale or something like that.
> > > Edit: Also here are my YAMLs for anyone who wants them. I added a bunch of other stuff. [debugger_method.yaml](https://gist.github.com/carbon198/7e6b3854d7dbc426f9b0a7c851bee871) [debugger_method_vars.h](https://gist.github.com/carbon198/4cb35b1d102653c1f52bcfcc5cc3872c)
> > 
> > 
> > I'm interested in kicking the unit back into F* with an ESP32.
> 
> Have you tried using this follow me feature? I think it should kick it back to F. If not I can test something else out.

Not sure why I didn't just give it a shot. Looks good, it's working as expected. Thank you!

carbon198 commented:
> > > > I've updated the branch to fix the celcius issue--the set temp will now remain in F. If you already switched to C you will need to coerce it back with the wall controller, or you can message me and I can tell you how to do with with your esp32 device. There's also now a toggle for Follow Me so you can turn it off if your temperature sensor gets stale or something like that.
> > > > Edit: Also here are my YAMLs for anyone who wants them. I added a bunch of other stuff. [debugger_method.yaml](https://gist.github.com/carbon198/7e6b3854d7dbc426f9b0a7c851bee871) [debugger_method_vars.h](https://gist.github.com/carbon198/4cb35b1d102653c1f52bcfcc5cc3872c)
> > > 
> > > 
> > > I'm interested in kicking the unit back into F* with an ESP32.
> > 
> > 
> > Have you tried using this follow me feature? I think it should kick it back to F. If not I can test something else out.
> 
> Not sure why I didn't just give it a shot. Looks good, it's working as expected. Thank you!

Awesome! I assumed that's what it would do since the other version from other projects forced to C, but I hadn't tested that it actually forced F. Good to know, thanks!

BTW if anyone stumbles into this thread looking for a Celsius version, just let me know.
